Prepare v4.2.0 release (#536)
grafana/azure-data-explorer-datasource: CHANGELOG.md

# Change Log

## [4.2.0]

This release revamps the plugin query builder:

> - New feature: It's now possible to filter columns to query, improving the performance of queries.

Reviewer: ```suggestion
- New feature: It's now possible to filter columns within a query, improving the performance of queries.
```

- The query preview includes syntax highlighting for Kusto.

> - The rest of components has been refactored to match the latest Grafana UI.

Reviewer: nit:
```suggestion
- All other components have been refactored to match the latest Grafana UI.
```
